pkg/kubectl/cmd/config/config.go

```diff
@@ -225,6 +225,9 @@ func ModifyConfig(configAccess ConfigAccess, newConfig clientcmdapi.Config, rela
 
 			configToWrite := getConfigFromFileOrDie(destinationFile)
 			t := *cluster
+			if configToWrite.Clusters == nil {
```

Comment:
:+1: I'm happy to do a follow on PR adding unittests for this if that works for @pmorie